meson.build

```diff
@@ -55,15 +56,20 @@ else
     endif
 endif
 
+if get_option('default_library') != 'static'
+    extra_args += '-DINI_SHARED_LIB'
+endif
+
 #### inih ####
 inc_inih = include_directories('.')
 
 lib_inih = library('inih',
     ['ini.c'],
     include_directories : inc_inih,
-    c_args : arg_static,
+    c_args : [arg_static, extra_args, '-DINI_SHARED_LIB_BUILDING'],
```

Comment:
This option is not *needed* except on Windows / MSVC, anyway you can "probably" use add_project_arguments() for this.

Reply:
By the way I noticed that you can pass `[arg_static, extra_args]` to `library.c_args` but not `declare_dependency.compile_args` (it works with `arg_static + extra_args` though) - _"List item must be one of <class 'str'>, not <class 'list'>"_

Reply:
Some places do implicit flattening at time of use, some do not. Personally I find implicit flattening makes me nervous.